pkg/resource/bucket/hook.go

```diff
@@ -0,0 +1,171 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package bucket
+
+import (
+	"context"
+
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	svcsdk "github.com/aws/aws-sdk-go/service/s3"
+)
+
+func (rm *resourceManager) syncPutFields(
+	ctx context.Context,
+	r *resource,
+) (synced *resource, err error) {
+	// Copy resource for later patching
+	synced = &resource{r.ko.DeepCopy()}
+
+	diff, err := rm.diffLogging(ctx, synced)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(diff.Differences) > 0 {
+		rm.syncLogging(ctx, synced)
+	}
+
+	return synced, nil
+}
+
+func (rm *resourceManager) diffLogging(
+	ctx context.Context,
+	r *resource,
+) (*ackcompare.Delta, error) {
+	desired := r.ko.Spec.Logging
+
+	input := &svcsdk.GetBucketLoggingInput{
+		Bucket: r.ko.Spec.Name,
+	}
+	latest, err := rm.sdkapi.GetBucketLoggingWithContext(ctx, input)
+	if err != nil {
+		return nil, err
+	}
+
+	delta := ackcompare.NewDelta()
```

Comment:
instead of creating a separate delta object here, you could change this function signature to accept a delta object and place a call to `diffLogging` in the `delta_pre_compare` hook:

https://github.com/aws-controllers-k8s/code-generator/blob/cae7cb2cccda567eab55c01873f89274c8f82b60/templates/pkg/resource/delta.go.tpl#L22-L24

and pass in the `delta` variable from there.